app/controllers/quotes_controller.rb

```diff
@@ -1,14 +1,29 @@
 class QuotesController < ApplicationController
+  before_action :authenticate_user!
   def show
     @quote = Quote.find params[:id]
   end
 
+  def edit
+    @quote = current_user.quotes.find params[:id]
+  end
+
+  def update
+    @quote = current_user.quotes.find params[:id]
+    if @quote.update quote_params
+      redirect_to @quote
+    else
+      render :edit
+    end
+  end
+
   def new
     @quote = Quote.new
   end
 
   def create
     @quote = Quote.new quote_params
+    @quote.user = current_user
```

Comment:
I think the problem comes up in that author and user can be used interchangeably. An author could be the user that created the record,  where the user could be the person who said it and vica versa. created_by and quoted_person/user or the like might be a bit easier to understand.